api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -302,6 +312,22 @@ func (m *AzureManagedControlPlane) Validate(cli client.Client) error {
 	return kerrors.NewAggregate(errs)
 }
 
+func (m *AzureManagedControlPlane) validateDNSPrefix(_ client.Client) error {
+	if m.Spec.DNSPrefix == nil {
+		return nil
+	}
+
+	// Regex pattern for DNS prefix validation
+	// 1. Between 1 and 54 characters long: {1,54}
+	// 2. Alphanumerics and hyphens: [a-zA-Z0-9-]
+	// 3. Start and end with alphanumeric: ^[a-zA-Z0-9].*[a-zA-Z0-9]$
+	regex := regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9-]{0,52}[a-zA-Z0-9]$`)
+	if regex.MatchString(ptr.Deref(m.Spec.DNSPrefix, "")) {
+		return nil
+	}
+	return errors.New("DNSPrefix is invalid")
```

Comment:
Including the regex in this message seems worthwhile so users have a bit more context as to why it's invalid.

Reply:
Or could this use [`// +kubebuilder:validation:Pattern`](https://book.kubebuilder.io/reference/markers/crd-validation.html)?

Reply:
@nojnhuh updated changes based on your first comment.